lib/tasks/after_tasks.rake

```diff
@@ -631,6 +631,28 @@ namespace :After do
     end
     puts && STDOUT.flush
   end
+
+  desc "Add user_id to existing logged in kudos without user_ids"
+  task(add_user_id_to_kudos: :environment) do
+    kudos_to_update = Kudo.where("pseud_id IS NOT NULL AND user_id IS NULL")
+    kudos_count = kudos_to_update.count
+    total_batches = kudos_count / 1000
+    puts "Total number of kudos to check: #{kudos_count}"
+
+    kudos_to_update.find_in_batches.with_index do |batch, index|
+      batch_number = index + 1
+      progress_msg = "Batch #{batch_number} of about #{total_batches} complete"
+      batch.each do |kudos|
+        # Skip unless the pseud and user exist.
+        next unless kudos&.pseud&.user
+
+        kudos.user_id = kudos.pseud.user_id
+        kudos.save if kudos.changed?
```

Comment:
That's a good point about co-creators. 😬 It would almost be nice to clean those up at the end of this as well, since we're already telling users they may lose kudos as a result of these changes, but there's nothing to stop it from happening in the future, so we'll just 🙈